### PR TITLE
Simplified rolling aggregate

### PR DESCRIPTION
**Pull request by ryzhyk**
See individual commits for details.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -59,7 +59,7 @@ jobs:
           - { rust: stable, target: i686-pc-windows-msvc, os: windows-latest }
           - { rust: beta, target: x86_64-unknown-linux-gnu, os: ubuntu-latest }
           - {
-              rust: nightly,
+              rust: nightly-2022-11-10,
```
Comment (mbudiu-vmw) [inline .github/workflows/main.yml:62]: why test on nightly?

File: benches/fraud.rs [hunk 2]
```diff
@@ -195,11 +195,9 @@ struct Args {
 type Weight = i32;
 
 type EnrichedTransactions = OrdIndexedZSet<(F64, i64), (Transaction, Demographics), Weight>;
-type AverageSpendingPerWeek =
-    OrdPartitionedIndexedZSet<F64, i64, (F64, Option<Avg<F64, i32>>), Weight>;
-type AverageSpendingPerMonth =
-    OrdPartitionedIndexedZSet<F64, i64, (F64, Option<Avg<F64, i32>>), Weight>;
-type TransactionFrequency = OrdPartitionedIndexedZSet<F64, i64, (F64, Option<i32>), Weight>;
+type AverageSpendingPerWeek = OrdPartitionedIndexedZSet<F64, i64, Option<Avg<F64, i32>>, Weight>;
```
Comment (mbudiu-vmw) [inline benches/fraud.rs:198]: you didn't fix the SQL... #239

File: src/operator/time_series/rolling_aggregate.rs [hunk 3]
```diff
@@ -391,23 +390,21 @@ where
...
                                     delta_cursor.key().clone(),
-                                    (
-                                        *input_range_cursor.key(),
-                                        (input_range_cursor.val().clone(), agg.clone()),
-                                    ),
+                                    (*input_range_cursor.key(), agg),
                                 ),
                                 HasOne::one(),
                             ));
+                            break;
```
Comment (mbudiu-vmw) [inline src/operator/time_series/rolling_aggregate.rs:407]: I find this surprising

**Review by mbudiu-vmw — APPROVED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (ryzhyk) [reply .github/workflows/main.yml:62]: to catch issues that we'll run into when nightly becomes stable.

Comment (ryzhyk) [reply .github/workflows/main.yml:62]: also, things like sanitizers only work on nightly

File: src/operator/time_series/rolling_aggregate.rs (hunk 3 above)
Comment (ryzhyk) [reply src/operator/time_series/rolling_aggregate.rs:407]: As soon as we encounter an input record with non-zero weight, we know that we need to compute the sliding window aggregate for the current timestamp.  We don't care about the actual value or the number of different values, so we can break out of the loop.

**Pull request merged**